python/ray/experimental/state.py

```diff
@@ -562,8 +571,10 @@ def micros(ts):
         def micros_rel(ts):
             return micros(ts - start_time)
 
-        task_profiles = self.task_profiles(start=0, end=time.time())
-        task_table = self.task_table()
```

Comment:
This was also a full table scan.

Reply:
There is actually a way to get the best of both worlds with the SCAN [1] class of functions from redis (I guess that's what you mean with the TODO above, do you want to make it a little more precise?)

[1] https://redis.io/commands/scan

Reply:
SCAN won't do it, but MGET might. I think the refactoring is a bit nontrivial but I updated the TODO.

Reply:
You are right, sounds good!